### DISCUSSION TITLE
Accessing the values of the default and custom configuration fields

### DISCUSSION THREAD

**Discussion by szabgab** [rwf2/Rocket#2709] [Questions] [answered]
I admit I did not totally understand the [page about configuration](https://rocket.rs/v0.5/guide/configuration/), but I did some experiments and wrote [this article](https://rust.code-maven.com/rocket-access-custom-configuration)
I'd appreciate a review for my article.

**Comment by SergioBenitez** [comment, marked as answer]
You absolutely should not do the following in a route handler:

```rust
    let custom_a: String = rocket::Config::figment().extract_inner("custom_a").unwrap_or(String::from("some default"));
```

Calling `figment()` is expensive as it reads all of the files and does all of deserialization.

Instead, use `AdHoc::config()` to extract the configuration in a typed fashion at launch time, then simply read it in a route:

```rust
use rocket::{State, fairing::AdHoc};

#[get("/")]
fn custom(config: &State<MyConfig>) -> String {
    config.custom_a.clone()
}

#[derive(Deserialize)]
struct MyConfig {
    custom_a: String,
}

#[launch]
fn rocket() -> _ {
    rocket::build()
        .mount("/", routes![custom])
        .attach(AdHoc::config::<Config>())
}
```

**Reply by szabgab** [reply]
Thanks for your reply. Finally I managed to update my blog post. btw there is a typo in your response in `AdHoc::config::<Config>()` it should be MyConfig.
